pkg/provider/azure_storageaccount.go

```diff
@@ -274,6 +315,97 @@ func (az *Cloud) EnsureStorageAccount(accountOptions *AccountOptions, genAccount
 	return accountName, accountKey, nil
 }
 
+func (az *Cloud) createPrivateEndpoint(ctx context.Context, accountName string, privateEndpointName string) error {
+	klog.V(5).Infof("Creating private endpoint(%s) for account (%s)", privateEndpointName, accountName)
+	subnet, rerr := az.SubnetsClient.Get(ctx, az.resourceGroup, az.cloud.VnetName, az.cloud.SubnetName, "")
+	if rerr != nil {
+		return rerr.Error()
+	}
+
+	storageAccount, rerr := az.StorageAccountClient.GetProperties(ctx, az.ResourceGroup, accountName)
+	if rerr != nil {
+		return rerr.Error()
+	}
+
+	// Disable the private endpoint network policies before creating private endpoint
+	subnet.SubnetPropertiesFormat.PrivateEndpointNetworkPolicies = network.VirtualNetworkPrivateEndpointNetworkPoliciesDisabled
+	rerr = az.SubnetsClient.CreateOrUpdate(ctx, az.ResourceGroup, az.VnetName, az.SubnetName, subnet)
+	if rerr != nil {
+		return rerr.Error()
+	}
+
+	//Create private endpoint
+	privateLinkServiceConnectionName := accountName + "-PrivateLinkServiceConn"
```

Comment:
Some of the suffices are quite long. Are endanger of exceeded resource name lengths if given a storage account name of maximum length?

Reply:
need to check the maximum length

Reply:
Looks like it's 80 characters for most of the cases: https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/resource-name-rules#microsoftnetwork

I'll try to make the names shorter